app-advanced-features/app/src/main/java/com/goldenraven/devkitwallet/data/Wallet.kt

```diff
@@ -89,60 +84,45 @@ object Wallet {
+        Log.i(TAG, "Loading existing wallet with descriptor: ${initialWalletData.descriptor}")
+        Log.i(TAG, "Loading existing wallet with change descriptor: ${initialWalletData.changeDescriptor}")
         initialize(
-            externalDescriptor = initialWalletData.externalDescriptor,
-            internalDescriptor = initialWalletData.internalDescriptor,
+            descriptor = Descriptor(initialWalletData.descriptor, Network.TESTNET),
+            changeDescriptor = Descriptor(initialWalletData.changeDescriptor, Network.TESTNET),
         )
     }
 
-    fun recoverWallet(mnemonic: String) {
-        val bip32RootKey: DescriptorSecretKey = DescriptorSecretKey(
-            network = Network.TESTNET,
-            mnemonic = mnemonic,
-            password = ""
-        )
-        val externalDescriptor: String = createExternalDescriptor(bip32RootKey)
-        val internalDescriptor: String = createInternalDescriptor(bip32RootKey)
+    fun recoverWallet(recoveryPhrase: String) {
+        val mnemonic = Mnemonic.fromString(recoveryPhrase)
+        val bip32ExtendedRootKey = DescriptorSecretKey(Network.TESTNET, mnemonic, null)
+        val descriptor: Descriptor = Descriptor.newBip84(bip32ExtendedRootKey, KeychainKind.EXTERNAL, Network.TESTNET)
+        val changeDescriptor: Descriptor = Descriptor.newBip84(bip32ExtendedRootKey, KeychainKind.INTERNAL, Network.TESTNET)
         initialize(
-            externalDescriptor = externalDescriptor,
-            internalDescriptor = internalDescriptor,
+            descriptor = descriptor,
+            changeDescriptor = changeDescriptor,
         )
-        Repository.saveWallet(path, externalDescriptor, internalDescriptor)
-        Repository.saveMnemonic(mnemonic)
+        Repository.saveWallet(path, descriptor.asStringPrivate(), changeDescriptor.asStringPrivate())
+        Repository.saveMnemonic(mnemonic.asString())
     }
 
-    fun createTransaction(
-        recipientList: MutableList<Recipient>,
-        feeRate: Float,
-        enableRBF: Boolean
-    ): PartiallySignedBitcoinTransaction {
-        // technique 1 for adding a list of recipients to the TxBuilder
-        // var txBuilder = TxBuilder()
-        // for (recipient in recipientList) {
-        //     txBuilder  = txBuilder.addRecipient(address = recipient.first, amount = recipient.second)
-        // }
-        // txBuilder = txBuilder.feeRate(satPerVbyte = fee_rate)
-
-        // technique 2 for adding a list of recipients to the TxBuilder
-        var txBuilder = recipientList.fold(TxBuilder()) { builder, recipient ->
-            builder.addRecipient(recipient.address, recipient.amount)
-        }
-        if (enableRBF) {
-            txBuilder = txBuilder.enableRbf()
-        }
-        return txBuilder.feeRate(feeRate).finish(wallet)
+    fun createTransaction(recipientList: MutableList<Recipient>, amount: ULong, fee_rate: Float): TxBuilderResult {
```

Comment:
I think you've copied the code from the simple wallet but notice that one of the features of the advanced wallet is the ability to send to multiple recipients (that's why the recipient argument is a list).